src/components/site-form.tsx

```diff
@@ -475,9 +487,15 @@ export const SiteForm = ( {
 											</div>
 										) }
 
-										{ setUseCustomDomain && setCustomDomain && (
-											<div className="text-a8c-gray-50 text-xs mt-2">
-												{ __( 'Your system password will be required to set up the domain.' ) }
+										{ useCustomDomain && setEnableSSL && (
+											<div className="flex items-center gap-2 mt-4">
+												<input
+													type="checkbox"
+													id="enable-ssl"
+													checked={ enableSSL }
+													onChange={ ( e ) => setEnableSSL( e.target.checked ) }
+												/>
+												<label htmlFor="enable-ssl">{ __( 'Enable SSL' ) }</label>
 											</div>
 										) }
```

Comment:
It would be really cool to enable this by default if the root certificate is trusted on the system. Do you think that's a good fit for this PR, @youknowriad, or should we pursue that separately?

Reply:
Happy to try that. Let's keep it separate